### DISCUSSION TITLE
Lazy Hooks

### DISCUSSION THREAD

**Discussion by emehmet** [refinedev/refine#2397] [Ideas]
Hi everyone,

All hook structures are configured to run when the component is rendered. But sometimes the data needs to be called again with the parameters to be given after a button is pressed. This can be done by adding the lazy property to all hook structures. Or, new components such as useTableLazy, useListLazy can be included. 
I needed this while developing code now and I'm looking at how to do it.

**Comment by omeraplak** [comment]
Hey @emehmet ,
Thank you for contacting us! Would the `refetch` function work for you?
https://refine.dev/docs/faq/#how-can-i-refetch-data
or
https://refine.dev/docs/faq/#how-can-i-ensure-a-query-is-only-runned-after-a-certain-variable-is-available-and-not-on-load

**Comment by emehmet** [comment]
Hi @omeraplak ,
Thank you for your solution.

I tried these structures but could not reach the result. After collecting data from useForm, I call refetch with button, but I couldn't find how to pass new variables to useTable.

**Reply by omeraplak** [reply]
Unfortunately, with react-query this is not very easy. You can get around this by keeping the Variables in state

edit: #2407

**Comment by emehmet** [comment]
I am trying to list the data with a table at the end from the data I collected using useStepsForm. useStepsForm queries are Graphql.mutation but I need to send Graphql.query query to server.

**Reply by omeraplak** [reply]
Hey @emehmet ,
You should be able to override `useStepsForm`'s `submit` function, but apparently, that doesn't work. Can you create a separate issue for this?

edit: #2407
